examples/arpscan/arpscan.go

```diff
@@ -176,13 +176,13 @@ func writeARP(handle *pcap.Handle, iface *net.Interface, addr *net.IPNet) error
 func ips(n *net.IPNet) (out []net.IP) {
 	num := binary.BigEndian.Uint32([]byte(n.IP))
 	mask := binary.BigEndian.Uint32([]byte(n.Mask))
-	num &= mask
-	for mask < 0xffffffff {
+	net := num & mask
+	broadcast := net | ^mask
+	for net++; net < broadcast-1; {
 		var buf [4]byte
-		binary.BigEndian.PutUint32(buf[:], num)
+		net++
```

Comment:
rather than this, how abut make line 181 be:

for net++; net<broadcast; net++ {
  ...
}

Reply:
Right! Missed that completely, thanks.